### ISSUE TITLE
nghttp2 client support

### ISSUE THREAD

**Issue by nijx** [tatsuhiro-t/nghttp2#238]
Hi, looks interesting library i'm currently testing examples from nghttp2 yet so far i only able to execute SSL requests does it support regular http ? if so please could you inform how it can be coded a simple client.

**Comment by tatsuhiro-t** [comment]
Assuming "regular http" as "cleartext HTTP/2", and you are talking about libevent-client.c in examples directory, quick hack to send request in cleartext HTTP/2 is replace bufferevent_openssl_socket_new with bufferevent_socket_new, and enable EV_READ for that bev:

```diff
diff --git a/examples/libevent-client.c b/examples/libevent-client.c
index 17ed94b..29d6b9a 100644
--- a/examples/libevent-client.c
+++ b/examples/libevent-client.c
@@ -486,9 +486,9 @@ static void initiate_connection(struct event_base *evbase, SSL_CTX *ssl_ctx,
   SSL *ssl;
 
   ssl = create_ssl(ssl_ctx);
-  bev = bufferevent_openssl_socket_new(
-      evbase, -1, ssl, BUFFEREVENT_SSL_CONNECTING,
-      BEV_OPT_DEFER_CALLBACKS | BEV_OPT_CLOSE_ON_FREE);
+  bev = bufferevent_socket_new(evbase, -1,
+                               BEV_OPT_DEFER_CALLBACKS | BEV_OPT_CLOSE_ON_FREE);
+  bufferevent_enable(bev, EV_READ);
   bufferevent_setcb(bev, readcb, writecb, eventcb, session_data);
   rv = bufferevent_socket_connect_hostname(bev, session_data->dnsbase,
                                            AF_UNSPEC, host, port);
```

**Comment by nijx** [comment]
kind regards, @tatsuhiro-t  i will test it and inform you back. :+1:

**Comment by nijx** [comment]
Hi, i have been re-modifying code yet i end up with this output of libevent-client example:
>ubuntu@ubuntu:~/Downloads/nghttp2/examples$ ./libevent-client http://www.google.com
Connected
Request headers:
:method: GET
:scheme: http
:authority: www.google.com
:path: 

Here is a source code of the modified examples [Gist] (https://gist.github.com/0b89d1bb4701b4c746a9.git)

correct me if i missed in coding. Thanks

**Comment by tatsuhiro-t** [comment]
It was a bug in libevent-client.c.  I have fixed it in 0cb4750

**Comment by tatsuhiro-t** [comment]
Please reopen if you still have this issue.
